src/frontend/packages/suse-extensions/src/custom/kubernetes/services/kubernetes.analysis.service.ts

```diff
@@ -0,0 +1,305 @@
+    const requestArgs = {
+      headers,
+    };
+
+    const start = this.http.post(url, body, requestArgs).pipe(
+      map(response => {
+        return response;
+      }),
+      catchError((e, c) => {
+        console.log('Error occurred');
+        console.log(e);
+        const msg = { firstLine: 'Failed to run Analysis Report'};
+        return of(false);
+      })
+    );
+
+    start.subscribe(a => {
+      const type = id.charAt(0).toUpperCase() + id.substring(1);
+      let msg;
+      if (app) {
+        msg = `${type} analysis started for workload '${app}'`;
+      } else if (namespace) {
+        msg = `${type} analysis started for namespace '${namespace}'`;
+      } else {
+        msg = `${type} analysis started for the Kubernetes cluster`;
+      }
+      const ref = this.snackBar.open(msg, 'View', { duration: 5000 });
+      ref.onAction().subscribe(() => {
+        this.store.dispatch(new RouterNav({ path: ['kubernetes', endpointID, 'analysis'] }));
+      });
+      this.refresh();
+    });
+  }
+
+  public getLatestCheck(endpointID: string, path: string): Observable<boolean> {
+    return this.getLatestObservable(endpointID, path, true).pipe(
+      map(response => response !== false),
+    );
+  }
+  public getLatest(endpointID: string, path: string): Observable<any> {
+
+    const start = this.getLatestObservable(endpointID, path, false).pipe(
+      map(response => {
+        this.processReport(response);
+        return response;
+      }),
+      catchError((e, c) => {
+        console.log('Error occurred');
+        console.log(e);
+        const msg = { firstLine: 'Failed to run Analysis Report'};
+        return of(false);
+      })
+    );
+
+    return start;
+  }
+
+  private getLatestObservable(endpointID: string, path: string, checkExists = false): Observable<any> {
+    const proxyAPIVersion = environment.proxyAPIVersion;
+    const url = `/pp/${proxyAPIVersion}/analysis/latest/${endpointID}/${path}`;
+    const headers = new HttpHeaders({});
+    const requestArgs = {
+      headers,
+    };
+
+    let req;
+    if (checkExists) {
+      req = this.http.head(url, requestArgs);
+    } else {
+      req = this.http.get(url, requestArgs);
+    }
+
+    return req.pipe(
+      catchError((e, c) => {
+        console.log('Error occurred');
+        console.log(e);
+        const msg = { firstLine: 'Failed to run Analysis Report'};
+        return of(false);
+      })
+    );
+
+  }
+
+  private processReport(report: any) {
+    // Check the path of the report
+    if (report.path.split('/').length !== 2) {
+      return;
+    }
+
+    switch (report.format) {
+      case 'popeye':
+        const helper = new PopeyeReportHelper(report);
+        helper.map();
+        break;
+      case 'kubescore':
+        const kubeScoreHelper = new KubeScoreReportHelper(report);
+        kubeScoreHelper.map();
+        break;
+      default:
+        console.log('Do not know how to handle this report type');
```

Comment:
This should be console.warning or similar, same with all error handling in this service